gqlalchemy/how-to-guides/query-builder.md

```diff
@@ -0,0 +1,1498 @@
+  <TabItem value="cypher">
+
+```cypher
+MATCH (c:Country) WHERE c.country_name = 'Germany' SET c += {population: '85000000'};
+```
+
+</TabItem>
+</Tabs>
+
+All the properties in the map (value of the `literal` argument) that are on a graph object will be updated. The properties that are not on a graph object but are in the map will be added. Properties that are not present in the map will be left as is.
+
+## Filter data
+
+You can use the methods [`where()`](/docs/gqlalchemy/reference/query_builders/declarative_base#where), [`where_not()`](/docs/gqlalchemy/reference/query_builders/declarative_base#where_not), [`or_where()`](/docs/gqlalchemy/reference/query_builders/declarative_base#or_where),
+[`or_where_not()`](/docs/gqlalchemy/reference/query_builders/declarative_base#or_where_node), [`and_where()`](/docs/gqlalchemy/reference/query_builders/declarative_base#and_where), [`and_where_not()`](/docs/gqlalchemy/reference/query_builders/declarative_base#and_where_not), [`xor_where()`](/docs/gqlalchemy/reference/query_builders/declarative_base#xor_where) and
+[`xor_where_not()`](/docs/gqlalchemy/reference/query_builders/declarative_base#xor_where_not) to construct queries that will filter data.
+
+
+
+### Filter data by property comparison
+
+To **filter data by comparing properties** of two nodes, run the following code:
+
+<Tabs
+defaultValue="gqlalchemy"
+values={[
+{label: 'GQLAlchemy', value: 'gqlalchemy'},
+{label: 'Cypher', value: 'cypher'}
+]}>
+<TabItem value="gqlalchemy">
+
+```python
+from gqlalchemy import match
+from gqlalchemy.query_builders.memgraph_query_builder import Operator
+
+results = list(
+    match()
+    .node(labels="Person", variable="p1")
+    .to(relationship_type="FRIENDS_WITH")
+    .node(labels="Person", variable="p2")
+    .where(item="p1.name", operator=Operator.LESS_THAN, expression="p2.name")
+    .return_()
+    .execute()
+)
+
+print(results)
+```
+
+  </TabItem>
+  <TabItem value="cypher">
+
+```cypher
+MATCH (p1:Person)-[:FRIENDS_WITH]->(p2:Person) WHERE p1.name < p2.name RETURN *;
+```
+
+  </TabItem>
+</Tabs>
+
+Keyword arguments that can be used in filtering methods are `literal` and `expression`. Usually we use `literal` for property values and `expression` for property names and labels. That is because property names and labels shouldn't be quoted in Cypher statements. 
+
+:::info
+You will probably see the `GQLAlchemySubclassNotFoundWarning` warning. This happens if you did not define a Python class which maps to a graph object in the database. To do that, check the [object graph mapper how-to guide](/docs/gqlalchemy/how-to-guides/ogm). To ignore such warnings, you can do the following before query execution:
+
+```python
+from gqlalchemy import models
+
+models.IGNORE_SUBCLASSNOTFOUNDWARNING = True
+```
+:::
+
+
+
+Standard boolean operators like `NOT`, `AND`, `OR` and `XOR` are used in the
+Cypher query language. To have `NOT` within `WHERE` clause, you need to use
+`where_not()` method.
```

Comment:
wait... are they used or not? so you can have NOT but not with the where clause?